node/sync/src/block_sync.rs

```diff
@@ -220,7 +220,8 @@ impl<N: Network> BlockSync<N> {
 
         // If there are no block requests, but there are pending block responses in the sync pool,
         // then try to advance the ledger using these pending block responses.
-        if block_requests.is_empty() && !self.responses.read().is_empty() {
+        // This is only necessary in router mode because the gateway must sync via the certificate approach.
```

Comment:
```suggestion
        // Note: This condition is guarded by `mode.is_router()` because validators sync blocks
        // using another code path that updates both `storage` and `ledger` when advancing blocks.
```